packages/artifact/src/internal/upload-http-client.ts

```diff
@@ -359,58 +363,91 @@ export class UploadHttpClient {
     }
 
     let retryCount = 0
-    const retryLimit = getUploadRetryCount()
+    const retryLimit = getRetryLimit()
+
+    // Increments the current retry count and then checks if the retry limit has been reached
+    // If there have been too many retries, fail so the download stops
+    const incrementAndCheckRetryLimit = (
+      message?: IHttpClientResponse
+    ): boolean => {
+      retryCount++
+      if (retryCount > retryLimit) {
+        if (message) {
+          // eslint-disable-next-line no-console
+          console.log(message)
```

Comment:
If there's an eslint rule against this, we can just use `core.info`

Reply:
Not quite, `core.info` only takes a string, while `console.log()` can take any type of object, that's why in a lot of places I'm having to use `console.log()`. 

Doing `core.info(whatever.toString())` also doesn't cut it because certain types of objects like json strings become near unreadable

Reply:
Could you post an example of what you're trying to log?

It seems like an explicit error format or some even just `JSON.stringify` would be better than just throwing the error into `console.log`

Reply:
In this particular case, `message` is of type `http.IncomingMessage`. https://github.com/actions/http-client/blob/60c8242c8cbb55e4ac31434511874b68bad4afa8/index.ts#L68
It's not really an error format, but rather an object dump of what is part of the http call.

If there are more than 5 retryable status codes during a single file upload then who knows what could be going on which is why I output the entire message. It has headers, raw headers and a host of other information that doesn't get logged into Kusto which i've learned the hard way, here is an example (very long output, but it basically everything we could ever want to know) . https://github.com/actions/upload-artifact/issues/62#issuecomment-606562787

Reply:
That seems like a lot more info than is necessary. It would be cleaner to log the status code and the response from the server as an error. If you need the headers, then I'd recommend logging those separately in a debug statement. Logging the entire object seems like the wrong approach here.

Reply:
I did some refactoring and I added a new `displayHttpDiagnostics` method inside `utils`. You can see it [here](https://github.com/actions/toolkit/blob/a4a3055ab024d935749026095611024390d9c205/packages/artifact/src/internal/utils.ts#L222). I swapped out existing `console.log(response)` with this new method.

In action this is what it looks like:
![image](https://user-images.githubusercontent.com/16109154/78613978-8c5e9a80-786d-11ea-9620-1dfc27aca6af.png)

Test run with just to see what it looks like is here: https://github.com/konradpabjan/artifact-test/runs/565950616?check_suite_focus=true#step:5:20

There is some other information that could be useful that I can't display without doing a full `console.log`, but I _think_ this should be sufficient (debugging low level http errors really sucks).

I am using `core.info` instead of `core.debug` because this should only be called when an error is hit. Users should always see this regardless if they have have diagnostics on/off.